app/utils/testing.py

```diff
@@ -1,16 +1,37 @@
+from typing import Dict
+
+from django.db.models.base import ModelBase
+from django.http import HttpResponse
+from django.template.context import Context
 from django.urls import reverse
 from django.utils.http import urlencode
 
 
-def reverse_with_query(name, query):
+def reverse_with_query(viewname: str, query: Dict[str, str], *args, **kwargs):
     """
     Returns a reversed URL with query parameters
     """
-    return reverse(name) + "?" + urlencode(query)
+    return reverse(viewname, *args, **kwargs) + "?" + urlencode(query)
 
 
-def get_change_list_results(response):
+def get_change_list_results(response: HttpResponse):
     """
     Returns the change list results from a response
     """
     return list(response.context["cl"].result_list)
```

Comment:
Does HttpResponse have `context`, or do testing responses always have context? I'm just trying to work out if the type annotation you used is correct. I would imagine you are looking for a template response of some kind.

Reply:
I'll have a look into it!

Reply:
Yeah, it's a TemplateResponse. HttpResponse doesn't have context.